### ISSUE TITLE
Searching without whole words (by substring)?

### ISSUE THREAD

**Issue by mnalis** [ENT8R/NotesReview#80]
I'd want to filter my own notes by substring. 
eg. for example I created a lot of notes with text `test2`, `test20`, `test21`, `test22`, `test22b`, `test22c` etc.

I'd want to find **all** of them. However, entering `test2`  will show only `test2` note. 

Is it somehow possible to find notes where `test2` is only part of the word?

**Comment by Lee-Carre** [comment]
I'm curious as to the reason (assuming that there is one) for requiring the matching of whole words. At a guess, I would suspect to keep the load on the API low, but 🤷‍♂️.

I encountered this limitation when reviewing my notes marking unmapped recycling collection points (added in passing (because [local government data is awful](//www.gov.je/recycling)), for me to return and map them properly in future).

In order to capture both variations of the root-term
* recycling
* recycle

I [queried for `recycl`](//ent8r.github.io/NotesReview/?view=map&map=11%2F49.2%2F-2.1&apply-bbox=true&query=recycl). Nil results.

Querying for each of [`recycle`](//ent8r.github.io/NotesReview/?view=map&map=11%2F49.2%2F-2.1&apply-bbox=true&query=recycle) & [`recycling`](//ent8r.github.io/NotesReview/?view=map&map=11%2F49.2%2F-2.1&apply-bbox=true&query=recycling) yielded the notes I was seeking (I forget, now, which term yielded zero, and which yielded the notes; hence using a sub-string to match both / either; thankfully I had used the same variation / term consistently).

The trouble with this approach is that instead of a single query (to the NR site & OSM API) which would match either (and thus all the notes I was seeking), I ended up having to perform (at least) 3 queries.
Point being; if this is purposely implemented for efficiency reasons, then sometimes it has the opposite effect.

**Comment by tuxayo** [comment]
Another use case in mapping parties: we map surveillances cameras. And sometimes we write "camera" "cameras" or "cam" so substring matching would allow to find all our notes reliably.

Is local search still a thing? https://github.com/ENT8R/NotesReview/issues/46#issuecomment-612053285 Or does in fully depend on the API?

**Comment by ENT8R** [comment]
I'm currently doing a few experiments with an improved search functionality of the backend and a few things are already working, including that a single word is treated like a substring, so words which include the search term and any following characters will now be included in the results by default.

> I'm curious as to the reason (assuming that there is one) for requiring the matching of whole words.

There is no real reason besides that I used the default text search of the database to search for the specified query until now. When I switched to the new backend, it was simply the best option to get something up and running fast.

> Is local search still a thing? https://github.com/ENT8R/NotesReview/issues/46#issuecomment-612053285 Or does in fully depend on the API?

The local search was a rather cumbersome approach to bypass certain limitations by the OSM API back then, so this is fortunately not used anymore right now as all requests are made using the new backend.
